r/inst/build_arrow_static.sh

```diff
@@ -85,7 +87,7 @@ ${CMAKE} -DARROW_BOOST_USE_SHARED=OFF \
     -G ${CMAKE_GENERATOR:-"Unix Makefiles"} \
     ${SOURCE_DIR}
 
-${CMAKE} --build . --target install
+${CMAKE} --build . --target install -- -j $N_JOBS
```

Comment:
we were building single core, the build times for the linux binaries in r-binary-package is now down to 10min with sccache :rocket: